packages/react/src/hooks/utils/useInvalidateOnBlock.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react'
+import { QueryKey, useQueryClient } from 'react-query'
+
+import { useBlockNumber } from '../network-status'
+
+export function useInvalidateOnBlock({
+  enabled,
+  queryKey,
+}: {
+  enabled?: boolean
+  queryKey: QueryKey
+}) {
+  const { data: blockNumber } = useBlockNumber({
+    enabled,
+    watch: enabled,
+  })
+  const queryClient = useQueryClient()
+
+  const startBlock = React.useRef<number>()
+  React.useEffect(() => {
+    if (!enabled) return
+    if (blockNumber === startBlock.current) return
+    if (!startBlock.current) {
+      startBlock.current = blockNumber
+      return
+    }
+
+    queryClient.invalidateQueries(queryKey)
+  }, [blockNumber, enabled, queryClient, queryKey])
```

Comment:
If `useBlockNumber`s implementation was updated slightly, we could use the `onSuccess` callback instead of a separate effect here.

Reply:
Good idea! Updated, and now the implementation is more basic 👌